src/main/java/backend/resource/TurboLabel.java

```diff
@@ -211,4 +217,128 @@ public int hashCode() {
     public int compareTo(TurboLabel o) {
         return actualName.compareTo(o.getActualName());
     }
+
+    /**
+     * To return the TurboLabel in labels that matches labelName
+     * Assumption: the labelName matches exactly 1 TurboLabel
+     *
+     * @param labels
+     * @param labelName
+     * @return
+     */
+    public static TurboLabel getMatchingTurboLabel(List<TurboLabel> labels, String labelName) {
+        assert labels.stream()
+                .filter(label -> label.getActualName().equals(labelName))
+                .findFirst()
+                .isPresent();
+        return labels.stream()
+                .filter(label -> label.getActualName().equals(labelName))
+                .findFirst()
+                .get();
+    }
+
+    public static List<String> getLabelsNameList(List<TurboLabel> labels) {
+        return labels.stream()
+                .map(TurboLabel::getActualName)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the list of labels which name contains labelName
+     * @param repoLabels
+     * @param labelName
+     * @return
+     */
+    public static List<String> filterByName(List<String> repoLabels, String labelName) {
+        return repoLabels
+                .stream()
+                .filter(name -> Utility.containsIgnoreCase(getName(name), labelName))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the list of labels which group contains labelGroup
+     * @param repoLabels
+     * @param labelGroup
+     * @return
+     */
+    public static List<String> filterByGroup(List<String> repoLabels, String labelGroup) {
+        if (labelGroup.isEmpty()) return repoLabels;
+
+        return repoLabels
+                .stream()
+                .filter(name -> {
+                    if (hasGroup(name)) {
+                        return Utility.containsIgnoreCase(getGroup(name), labelGroup);
```

Comment:
So this can be a partial match too? In that case, need to mention in the header comment and name the method accordingly.